src/plugins/workflow_designer/src/WorkflowEditor.cpp

```diff
@@ -616,6 +617,32 @@ void SpecialParametersPanel::sl_datasetsChanged() {
     editor->commitDatasets(attrId, sets[attrId]);
 }
 
+void SpecialParametersPanel::sl_datasetRenamed(QPair<QString, QString>& oldNewNamePair) {
+    auto ctrl = dynamic_cast<AttributeDatasetsController*>(sender());
+    CHECK(ctrl != nullptr, );
+    CHECK(controllers.values().contains(ctrl), );
+    QString attrId = controllers.key(ctrl);
+    sets[attrId] = ctrl->getDatasets();
+    editor->commitDatasets(attrId, sets[attrId]);
+    for (Actor* actor : qAsConst(allActors)) {
+        const QStringList keys = actor->getParameters().keys();
+        for (const QString& key : qAsConst(keys)) {
+            URLAttribute* urlAttr = dynamic_cast<URLAttribute*>(actor->getParameter(key));
+            if (urlAttr == nullptr) {
+                continue;
+            }
+            QList<Dataset> datasetList = urlAttr->getAttributePureValue().value<QList<Dataset>>();
+            for (QList<Dataset>::iterator it(datasetList.begin()); it != datasetList.end(); it++) {
```

Comment:
Well... It's kinda work and correct, so I can say anything, but iterators? They are deprecated as hell. Just like this could be more obvious:
`for (const auto* dataset : qAsConst(datasetList)) {`

Reply:
We need to modify element in current container, so i choose iterator.